liota/dcc/vrops.py

```diff
@@ -150,9 +150,20 @@ def subscribe(self):
         pass
 
     def publish(self, metric):
-        timestamps = [t for t, _ in metric.values]
-        values = [v for _, v in metric.values]
+        met_cnt = metric.values.qsize()
+        if met_cnt == 0:
+            return
```

Comment:
Since we already talked about this, I am gonna be brief.
`dcc.publish` should take a list, e.g. `dcc.publish(self, metric_values)`, the list will be a list of tuples got from the queue (`metric.values`).
`metric.send_data` is gonna retrieve data from the queue (`self.values`). That means before it calls `self.dcc.publish` it should count the number of items in the queue, and do so many number of `self.values.get( )` to get exactly the same amount of numbers, and put these things into a list, and finally call `self.dcc.publish(that_list)`.
Then, dcc codes can stay ALMOST unchanged compared to the original implementation.